src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2744,7 +2748,9 @@ NetworkOPsImp::pubProposedTransaction(
     std::shared_ptr<STTx const> const& transaction,
     TER result)
 {
-    Json::Value jvObj = transJson(*transaction, result, false, ledger);
+    MultiApiJson jvObj(
+        {transJson(transaction, result, ledger, false, std::nullopt, 1),
+         transJson(transaction, result, ledger, false, std::nullopt, 2)});
```

Comment:
We are building different JSONs here, for different API versions. One still has `Amount` field and the other does not. In another PR (to unify transaction JSON output - which is built on top of this one) the differences are much larger - different name of JSON element for the transaction data, different location of `hash` element etc. In principle it is possible to generate one JSON element and then transform it from one version to another, but that would be very brittle. I think the best we can do is to do some kind of memoization, so building the transaction JSON second time takes fewer cycles.

Reply:
... needless to say, the way JSON population currently works (any part of code can do JSON), such memoization would cause not a small amount of churn, which I'd rather avoid.

Reply:
oh I just looked at your branch. Good idea with changing the type of `transJson` result, I think I am going to copy it.